### DISCUSSION TITLE
✅ 2FA support for SSH

### DISCUSSION THREAD

**Discussion by hoytler** [isontheline/pro.webssh.net#597] [Ideas]
I've attempted to use Google Authenticator with SSH and WebSSH seemingly doesn't let me connect when this is enabled.
I just receive a "failed to connect" error in the WebSSH app.
Please add 2FA support or at the least, support Google Authenticator, which is commonly used.

**Comment by isontheline** [comment]
Hello 👋 

2FA should already be working.
Have you enabled "force keyboard authentication"?

Could you provide verbose logs?

In order to enable verbose logging you can follow the steps :
1. iOS Settings
2. WebSSH
3. Settings
4. Log Level > Verbose
5. Restart WebSSH
6. Reproduce your action
7. Find the log files inside iOS > Files > WebSSH

**Reply by hoytler** [reply]
Yes, it is working. The magic fix for me was enabling "force keyboard auth" - I didn't realize this was a requirement.
Thanks for the quick response!

**Reply by isontheline** [reply]
You're welcome!

Thanks for your feedback 🙏

**Reply by jtbr** [reply]
Thanks, this worked for me as well. Maybe it could be mentioned in the error message?

**Reply by isontheline** [reply]
Thanks @jtbr for your feedback 🙏 I will improve the GUI in order to be more explicit
